pkg/controller/projectclaim/customresourceadapter_test.go

```diff
@@ -208,4 +210,74 @@ var _ = Describe("Customresourceadapter", func() {
 			})
 		})
 	})
+
+	Context("EnsureProjectClaimState()", func() {
+		var (
+			requestedState gcpv1alpha1.ClaimStatus
+			currentState   gcpv1alpha1.ClaimStatus
+		)
+		JustBeforeEach(func() {
+			projectClaim.Status.State = currentState
+		})
+
+		Context("when requested state is Pending", func() {
+			BeforeEach(func() {
+				requestedState = gcpv1alpha1.ClaimStatusPending
+			})
+
+			Context("when ProjectClaim state is not empty", func() {
+				BeforeEach(func() {
+					currentState = gcpv1alpha1.ClaimStatusReady
+				})
+				It("doesn't change the ProjectClaim state", func() {
+					adapter.EnsureProjectClaimState(requestedState)
```

Comment:
```suggestion
					err := adapter.EnsureProjectClaimState(requestedState)
					Expect(err).To(HaveOccurred())
```
if  we change the function to return an error, we could verify here.